azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -0,0 +1,310 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+"""
+This module is the requests implementation of Pipeline ABC
+"""
+from __future__ import absolute_import  # we have a "requests" module that conflicts with "requests" on Py2.7
+import contextlib
+import json
+import logging
+import os
+import platform
+import threading
+from typing import TYPE_CHECKING, cast, List, Callable, Iterator, Any, Union, Dict, Optional  # pylint: disable=unused-import
+import xml.etree.ElementTree as ET
+import warnings
+import types
+import re
+
+from azure.core import __version__  as azcore_version
+from .base import HTTPPolicy, SansIOHTTPPolicy
+from urllib3 import Retry  # Needs requests 2.16 at least to be safe
+
+from azure.core.exceptions import (
+    DecodeError,
+    raise_with_traceback
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HeadersPolicy(SansIOHTTPPolicy):
+    """A simple policy that sends the given headers
+    with the request.
+
+    This overwrite any headers already defined in the request.
+    """
+    def __init__(self, headers=None, **kwargs):
+        # type: (Mapping[str, str]) -> None
+        self.headers = headers or {}
+
+    def on_request(self, request, **kwargs):
+        # type: (PipelineRequest, Any) -> None
+        additional_headers = kwargs.pop('headers', {})
+        http_request = request.http_request
+        http_request.headers.update(self.headers)
+        http_request.headers.update(additional_headers)
+
+
+class UserAgentPolicy(SansIOHTTPPolicy):
+    _USERAGENT = "User-Agent"
+    _ENV_ADDITIONAL_USER_AGENT = 'AZURE_HTTP_USER_AGENT'
+
+    def __init__(self, base_user_agent=None, **kwargs):
+        # type: (Optional[str], bool) -> None
+        self.overwrite = kwargs.pop('user_agent_overwrite', False)
+        self.use_env = kwargs.pop('user_agent_use_env', True)
+
+        if base_user_agent is None:
+            self._user_agent = "python/{} ({}) azure-core/{}".format(
+                platform.python_version(),
+                platform.platform(),
+                azcore_version
+            )
+        else:
+            self._user_agent = base_user_agent
+
+    @property
+    def user_agent(self):
+        # type: () -> str
+        """The current user agent value."""
+        if self.use_env:
+            add_user_agent_header = os.environ.get(self._ENV_ADDITIONAL_USER_AGENT, None)
```

Comment:
There may be push-back on picking up this environment variable in our package.

Reply:
Yup, happy to keep/remove based on consensus.

Reply:
Why? It's completely generic?